bin/cheritest/cheritest.c

```diff
@@ -981,15 +981,14 @@ static const struct cheri_test cheri_tests[] = {
 	  .ct_desc = "check tags are not stored for tmpfile() MAP_SHARED pages",
 	  .ct_func = cheritest_vm_notag_tmpfile_shared,
 	  .ct_flags = CT_FLAG_SIGNAL | CT_FLAG_SI_CODE | CT_FLAG_SI_TRAPNO,
-	  .ct_signum = SIGPROT,
-	  .ct_si_code = PROT_CHERI_STORETAG,
-	  .ct_si_trapno = TRAPNO_CHERI,
+	  .ct_signum = SIGSEGV,
+	  .ct_si_code = SEGV_STORETAG,
 #ifdef __riscv
-	  .ct_xfail_reason = "CHERI-RISC-V doesn't implement CHERI PTE bits",
+	  .ct_si_trapno = EXCP_STORE_AMO_CAP_PAGE_FAULT,
 #else
-	  .ct_check_xfail = xfail_need_writable_non_tmpfs_tmp,
+	  .ct_si_trapno = TRAPNO_CHERI,
 #endif
-	},
+	  .ct_check_xfail = xfail_need_writable_tmp },
```

Comment:
This slipped through; the rest of it has gone.

Reply:
This is on purpose.  Once the PTE support was added, then both RISC-V and MIPS used the same ct_check_xfail of `xfail_need_writable_non_tmpfs_tmp`, but the followup change to strip tags on tmpfs then changed it to just use `xfail_need_writable_tmp`.

Reply:
Ah, right, yes. Your formatting needs fixing though.

Reply:
.. or not, that's the formatting we go with. Ew.

Reply:
Oh, yeah.  I found it a bit curious myself.